core/domain/user_services.py

```diff
@@ -850,19 +922,70 @@ def create_new_user(gae_id, email):
     Raises:
         Exception. If a user with the given gae_id already exists.
     """
-    user_settings = get_user_settings(gae_id, strict=False)
+    user_settings = get_user_settings_by_gae_id(gae_id, strict=False)
     if user_settings is not None:
-        raise Exception('User %s already exists.' % gae_id)
+        raise Exception(
+            'User %s already exists for gae_id %s.'
+            % (user_settings.user_id, gae_id))
 
     user_id = user_models.UserSettingsModel.get_new_id('')
     user_settings = UserSettings(
         user_id, gae_id, email, feconf.ROLE_ID_EXPLORATION_EDITOR,
         preferred_language_codes=[constants.DEFAULT_LANGUAGE_CODE])
+    _save_user_auth_details(UserAuth(user_id, gae_id))
     _save_user_settings(user_settings)
     create_user_contributions(user_id, [], [])
     return user_settings
 
 
+def _save_user_auth_details(user_auth):
```

Comment:
This is actually a better name. Could you name the domain object UserAuthDetails and the storage model UserAuthDetailsModel? Make sure to change variable/function names accordingly, too.

Reply:
Did the name change everywhere.